EC_finetune/agents.py

```diff
@@ -80,9 +80,20 @@ def freeze_adapters(self) -> None:
             param.requires_grad = False
         for param in self.receiver_reshaper.parameters():
             param.requires_grad = False
-        if self.sender.recurrent_unroll:
+        if self.sender.unroll == 'recurrent':
             for param in self.sender.lstm.parameters():
                 param.requires_grad = False
+        elif self.sender.unroll == 'transformer':
+            for param in self.sender.transformer.parameters():
+                param.requires_grad = False
+
+    def freeze_sender_decoder(self) -> None:
+        for param in self.sender.decoder.parameters():
+            param.requires_grad = False
+
+    def freeze_listener_encoder(self) -> None:
+        for param in self.receiver.encoder.parameters():
+            param.requires_grad = False
```

Comment:
This can be refactored: add a little method that's like `freeze_parameters(params):` which does the for loop, and then call that from each of these three methods, just with diff args for `params` (e.g. `self.sender.lstm`, `self.sender.transformer`, etc)

Reply:
good point!